File: samples/origami-simple-scala-app/app/models/vertices/Log.java
package models.vertices;

import java.util.List;

import play.Logger;
import play.data.validation.Constraints.Required;

import com.wingnest.play2.origami.GraphDB;
import com.wingnest.play2.origami.annotations.Index;

import views.txt.sql.*;

public class Log extends GraphVertexModelBase {

	@Required
	public String title;

	@Required
	@Index
	public String name;

	@Required
	public String logBody;

	public List<Comment> getComments() {
		final String query = getComments.render(this.getORID().toString()).body();
		Logger.debug("getComments : query = " + query);
		return GraphDB.synchQueryModel(query);
	}
}

File: samples/origami-simple-scala-app/app/models/vertices/Log.scala
package models.vertices

import java.util.List
import play.Logger
import com.wingnest.play2.origami.GraphDB
import com.wingnest.play2.origami.annotations.Index
import scala.annotation.meta.field

class Log extends GraphVertexModelBase {

  @field
  var title: String = null

  @(Index @field)
  var name: String = null

  @field
  var logBody: String = null

  def getComments(): List[Comment] = {
    var query = views.txt.sql.getComments.render(getORID().toString()).body
    GraphDB.synchQueryModel(query)
  }

}
